### ISSUE TITLE
bug: HDFS Storage flush error

### ISSUE THREAD

**Issue by yufansong** [risingwavelabs/risingwave#9147]
### Describe the bug

When trying to use `HDFS` as the storage object and flush content, hummock will meet some problems.  And also, when selecting HDFS as the storage, some e2e tests will fail.
![image](https://user-images.githubusercontent.com/33971064/231596447-fcb680ba-cf66-4482-a327-2165694719f3.png)

In my environment, when launched by `./risedev d`, flush is ok, e2e test is also ok. When opening HDFS feature, build, `./risedev d hdfs`, flush and e2e test will meet the problem above.

When I check the HDFS, there are some data already storage successfully.
![image](https://user-images.githubusercontent.com/33971064/231597218-c04d0504-88c1-46c0-b1cb-7a79e86235a3.png)

### To Reproduce

1. Use the branch `wcy/push_for_yufan` which already enables the hdfs feature. 
2. Launch the EC2 machine and try to build the branch and then `./risedev d hdfs`
3. Connect the database, write data, and flush it.

**Comment by wcy-fdu** [comment]
I believe this is some hdfs config issue, will check it later.

**Comment by Xuanwo** [comment]
Hi, can this issue be reproduced consistently or is it random?

---

Just a quick note, OpenDAL has built-in support for retrying operations in case of temporary errors. You can find more information about it at this link: https://docs.rs/opendal/latest/opendal/layers/struct.RetryLayer.html.

**Comment by wcy-fdu** [comment]
May due to the hdfs classpath is incorrect.
```
loadFileSystems error:
(unable to get stack trace for java.lang.NoClassDefFoundError exception: ExceptionUtils::getStackTrace error.)
hdfsBuilderConnect(forceNewInstance=0, nn=127.0.0.1:9000, port=0, kerbTicketCachePath=(NULL), userName=(NULL)) error:
(unable to get stack trace for java.lang.NoClassDefFoundError exception: ExceptionUtils::getStackTrace error.)
thread 'risingwave-main' panicked at 'called `Result::unwrap()` on an `Err` value: Unexpected (temporary) at  => uncategorized error, source: Unknown error 255 (os error 255)
```

**Comment by yufansong** [comment]
I think in the test case I mentioned, the database has already flushed part of the data into the system, which means the program should have already found the correct HDFS classpath.

**Comment by Xuanwo** [comment]
I think the problem has been resolved with https://github.com/apache/incubator-opendal/pull/2114. Kindly upgrade to opendal `v0.33.1` and try again.

**Comment by wcy-fdu** [comment]
> I think the problem has been resolved with [apache/incubator-opendal#2114](https://github.com/apache/incubator-opendal/pull/2114). Kindly upgrade to opendal `v0.33.1` and try again.

Thanks, let me try it later.

**Comment by wcy-fdu** [comment]
After a quick verification, I think hdfs works now. I will recheck this later, thanks for your work~ @Xuanwo 
<img width="1051" alt="image" src="https://user-images.githubusercontent.com/58715567/234336794-acde319e-0b3e-42d5-ab20-5d88ce4b287b.png">
